coordinator/src/main/java/com/tencent/rss/coordinator/ClientConfManager.java

```diff
@@ -0,0 +1,145 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.coordinator;
+
+import java.io.Closeable;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClientConfManager implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(ClientConfManager.class);
+
+  private final AtomicReference<Map<String, String>> clientConf = new AtomicReference<>();
+  private final AtomicLong lastCandidatesUpdateMS = new AtomicLong(0L);
+  private Path path;
+  private ScheduledExecutorService updateClientConfSES = null;
+  private FileSystem fileSystem;
+  private static final String WHITESPACE_REGEX = "\\s+";
+
+  public ClientConfManager(CoordinatorConf conf, Configuration hadoopConf) throws Exception {
+    if (conf.getBoolean(CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_ENABLED)) {
+      init(conf, hadoopConf);
+    }
+  }
+
+  private void init(CoordinatorConf conf, Configuration hadoopConf) throws Exception {
+    String pathStr = conf.get(CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_PATH);
+    this.path = new Path(pathStr);
+
+    this.fileSystem = CoordinatorUtils.getFileSystemForPath(path, hadoopConf);
+    if (!fileSystem.isFile(path)) {
+      String msg = String.format("Fail to init ClientConfManager, %s is not a file", path.toUri());
+      LOG.error(msg);
+      throw new IllegalStateException(msg);
+    }
+
+    FileStatus[] fileStatus = fileSystem.listStatus(path);
+    if (ArrayUtils.isEmpty(fileStatus)) {
+      throw new IllegalStateException("Fail to list file " + path);
+    }
+
+    int updateIntervalS = conf.getInteger(CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_UPDATE_INTERVAL_SEC);
+    updateClientConfSES = Executors.newSingleThreadScheduledExecutor(
+        new ThreadFactoryBuilder().setDaemon(true).setNameFormat("ClientConfManager-%d").build());
+    updateClientConfSES.scheduleAtFixedRate(
+        this::updateClientConf, 0, updateIntervalS, TimeUnit.SECONDS);
+  }
+
+  private void updateClientConf() {
+    try {
+      FileStatus[] fileStatus = fileSystem.listStatus(path);
+      if (!ArrayUtils.isEmpty(fileStatus)) {
+        long lastModifiedMS = fileStatus[0].getModificationTime();
```

Comment:
logic is a little trikcy.

Reply:
Update if the file is modified and ignore if empty content or file not exist.